### ISSUE TITLE
Remove `ember-rfc176-data` patch

### ISSUE THREAD

**Issue by buschtoens** [ember-polyfills/ember-cached-decorator-polyfill#1]
The required `Ember._memo` mapping for `{ memo } from '@glimmer/tacking'` has not yet been added to [`ember-rfc176-data`](https://github.com/ember-cli/ember-rfc176-data). This is why we use `patch-package` to patch this in, just so we can get our feet off the ground. For a "stable" release, we need to remove this hackery and land the required change upstream.

These commits need to be reverted:

- a94754369e38c19576bee8166609d0a5e3066540
- e35b96414dd164b4aeeb4bf3747e5bb2fddf12f3
- 4ac2e1ff07fa01f2d5c0b1aaf717613bb023cd1a

**Comment by buschtoens** [comment]
I opened https://github.com/ember-cli/ember-rfc176-data/pull/322 to add the mapping.

**Comment by buschtoens** [comment]
Unfortunately it seems that (at least with `yarn`) the `postinstall` script is not (properly) invoked and the patch for `ember-rfc-176-data` included in this package is not applied in the host project. To be honest, this is probably for the better any way. 😂 

I tried patching the package in `index.js` as well, like so:

```js
function patchRFC176Data() {
  const mapping = require("ember-rfc176-data");
  const hasMemoMapping = Boolean(mapping.find(
    (m) => m.module === "@glimmer/tracking" && m.export === "memo"
  ));
  if (hasMemoMapping) return;
  mapping.push({
    global: "Ember._memo",
    module: "@glimmer/tracking",
    export: "memo",
    deprecated: false,
  });
}
```

However, this does not work, as `ember-cli-babel` spawns its workers in child processes, which have their own `require.cache`. 🙁

**Comment by buschtoens** [comment]
Seems like another option is to add an `afterInstall` blueprint in which we ask the user, whether they would like us to setup up the `patch-package` hack in their repo.

Another, possibly much cleaner option could be adding this export to [`ignore`](https://github.com/ember-cli/babel-plugin-ember-modules-api-polyfill/blob/master/src/index.js#L70) and then transpiling it ourselves.

**Comment by mydea** [comment]
Hmm, when installing this locally via yarn, it worked, without applying the patch I believe. (at least the import did not work, I used it directly from `Ember._cached`). However, now on our CI on Github Actions, it is failing on the install step `yarn --frozen-lockfile` with:


```
[5/5] Building fresh packages...
error /.../node_modules/ember-cached-decorator-polyfill: Command failed.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
Exit code: 1
Command: patch-package
Arguments: 
Directory: /.../node_modules/ember-cached-decorator-polyfill
Output:
patch-package 6.2.2
Applying patches...
Error: Patch file found for package ember-rfc176-data which is not present at node_modules/ember-rfc176-data
```

I did nothing but `ember install` this addon, and it is weird that it works locally with yarn but not on CI. But I guess it is somehow related to this issue here.

**Comment by buschtoens** [comment]
Yeah, it seems that `patch-package` is behaving weirdly. :/

I would like to find some hack to completely avoid it. The most probable solution would be [disabling the transpilation](https://github.com/ember-cli/babel-plugin-ember-modules-api-polyfill/blob/fe9668c2d0ff236aea313e626e7f537b6a4a5208/src/index.js#L70) of the `@glimmer/tracking` import, and then either transpiling it ourselves (to avoid issues with `embroider`'s resolution system) or registering a faux `@glimmer/tracking` modules via `require.define()`.

I can push and release a fix, that would at least disable the `patch-package` stuff when this package is installed as a dependency. Then you can patch yourself or use `Ember._cached` and CI should work again.

**Comment by mydea** [comment]
That would work for me, at least :) So if that is OK, then I'd be all for that!

**Comment by buschtoens** [comment]
I've removed `patch-package` and replaced it with some `ember-cli-babel` h4xx in #12.

It's released in [`v0.1.0-3`](https://github.com/ember-polyfills/ember-cached-decorator-polyfill/releases/tag/v0.1.0-3). 🚀 

For reference I also raised https://github.com/ember-cli/babel-plugin-ember-modules-api-polyfill/issues/133 to improve the underlying infrastructure for future polyfills.
